http-server/src/main/java/io/micronaut/http/server/cors/CorsFilter.java

```diff
@@ -150,7 +151,7 @@ protected Optional<MutableHttpResponse<?>> handleRequest(HttpRequest request) {
                     }
                 }
 
-                if (preflight) {
+                if (preflight && RouteMatchUtils.findRouteMatch(request).isPresent()) {
```

Comment:
i don't think this is enough, what if this is a preflight for a POST request for example, but there's no GET controller. this will yield a 404 even though the actual cors request would succeed

Reply:
I think that you are right. I added new test case that covers that situation and I improved the solution.